apps/sms/views/conversation/js/conversation.js

```diff
@@ -2912,103 +2890,62 @@ var ConversationView = {
     new OptionMenu(params).show();
   },
 
-  onCreateContact: function conv_onCreateContact() {
-    InboxView.updateContactsInfo();
-    // Update Header if needed
-    if (Navigation.isCurrentPanel('thread')) {
-      ConversationView.updateHeaderData();
+  discardDraft: function conv_discardDraft() {
+    // If we were tracking a draft properly, update the Drafts object entry.
+    if (!this.draft) {
+      return;
     }
-  },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft
-    // properly update the Drafts object
-    // and InboxView entries
-    if (this.draft) {
-      Drafts.delete(this.draft).store();
-      if (Threads.active) {
-        Threads.active.timestamp = Date.now();
-        InboxView.updateThread(Threads.active);
-      } else {
-        InboxView.removeThread(this.draft.id);
-      }
-      this.draft = null;
+    // Update thread timestamp. Will be removed in bug 958105.
+    var thread = Threads.active;
+    if (thread) {
+      thread.timestamp = Date.now();
     }
+
+    Drafts.delete(this.draft).store();
```

Comment:
Maybe not in this patch, but we might need to move the store in delete to sycn Drafts.add without expose store, or move the store out of the add like Drafts.delete. I think we might delay the store action while batch deleting previously, but now we don't do this any more.